libraries/AP_HAL_Linux/CAN.h

```diff
@@ -0,0 +1,235 @@
+/* This code was taken from UAVCAN project, which uses The MIT License (MIT).
+
+   Copyright (c) 2014 Pavel Kirienko
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy of
+   this software and associated documentation files (the "Software"), to deal in
+   the Software without restriction, including without limitation the rights to
+   use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+   the Software, and to permit persons to whom the Software is furnished to do so,
+   subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+   FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+   COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+   IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+/* This code was modified to use with ArduPilot project, which uses GPL license.
```

Comment:
Is this how it works when MIT source gets bumped to GPL? There's both liceneses here. I have a feeling the MIT gets tripped and replaced, instead of appended, with the GPL. We should follow up on this. Definitely mentioning the other author/source is important so maybe this is correct (but maybe swapped with GPL at top)?

Reply:
We were asked to mention both original UAVCAN license and ArduPilot license.

Reply:
Yup, that's a fair ask. I'm asking what is the legally correct method that we're obligated to do. This might be correct as-is, I don't know.

Reply:
I'd suggest a line like this in files that have substantial content from Pavel from uavcan:

  See modules/uavcan/LICENSE for license details of the uavcan code from Pavel Kirenko